dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -78,7 +76,7 @@ sealed class TypedColumn[T, U](
     *
     * apache/spark
     */
-  def =!=(other: TypedColumn[T, U]): TypedColumn[T, Boolean] = withExpr {
+  def =!=[TT, W](other: TypedColumn[TT, U])(implicit w: With.Aux[T, TT, W]): TypedColumn[W, Boolean] = withExpr {
```

Comment:
Can we go a bit over the `With` type class? why do we need to do this?

Reply:
I answered in the main thread to not lose the message on rebase